pkg/controller/controller_binding.go

```diff
@@ -1596,55 +910,413 @@ func (c *controller) isServiceBindingReconciliationRetryDurationExceeded(binding
 	return false
```

Comment:
The `isServiceBindingReconciliationRetryDurationExceeded` function can be removed.